### DISCUSSION TITLE
go: could not create module cache: stat /root/go-workspace/pkg/mod: permission denied

### DISCUSSION THREAD

**Discussion by R1kk4-213** [projectdiscovery/nuclei#3860] [Q&A]
I Have got this issue when running 
`go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest` command
I have downloaded latest Go version for Linux 
"`┌──(kali㉿kali)-[~]
└─$ go version                                                        
go version go1.20.5 linux/amd64
                                 `
Please help

**Comment by MetzinAround** [comment]
That error, `go: could not create module cache: stat /root/go-workspace/pkg/mod: permission denied` is a system error, not a subfinder error. You don't seem to have the right permissions for your user. 

Just in case, make sure your GOPATH is in your $PATH. That might help, too. 

https://youtu.be/wtHcXA2l9c4?t=123

**Comment by Mzack9999** [comment]
@hoainam125 I've just tried with kali both in the docker and virtual box, and the default permissions for `kali` seems to be set correctly. I guess that you probably installed go as `root` (as you have the workspace in the `/root` folder) and are trying to run it as `kali`, which doesn't have enough permissions to write to `/root/go-workspace`

**Comment by krootca** [comment]
You can use this:

```bash
go env -w GOMODCACHE=${PWD}/CACHE_EXAMPLE/modcache GOCACHE=${PWD)/CACHE_EXAMPLE/buildcache
```

You can change the folder "CACHE_EXAMPLE" for other folder.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
